### PR TITLE
Adding a extra condition to restart_incomplete_calcution

### PR DESCRIPTION
**Pull request by AndresOrtegaGuerrero**
The new conditions ask for the end of SCF inner cycle with "Total energy:", This guarantees that there is *.wfn file in the directory

### CODE REVIEW

**Review by yakutovicha — COMMENTED** [review]
Please fix the test and request my review then 👍

**Comment by AndresOrtegaGuerrero** [comment]
I restored the initial conditions (remove my changes) , but somehow is not working , I get an error 402 (which i dont find). I assume now it should pass the test in github, yet i need to work on the solution

**Comment by AndresOrtegaGuerrero** [comment]
The issue seems that the `example_base_failed_restart.py` , checks if the process exit_status is `400`, however, there could be the case that in the first restart it successfully completes the calculation, then the exit code is `0` for the workchain , or is `402` which means that 2 attempts were not enough to complete the simulations. I include a modification to check the first calcjob to check if the restart is handled.

File: examples/workchains/example_base_energy_restart.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    structure = StructureData(
+        ase=ase.io.read(os.path.join(thisdir, "..", "files", "h2o.xyz"))
+    )
+
+    # Parameters.
+    parameters = Dict(
+        {
+            "GLOBAL": {
+                "RUN_TYPE": "ENERGY",
+                "WALLTIME": "00:00:05",  # Can't even do one geo opt step.
```
Comment (yakutovicha) [inline examples/workchains/example_base_energy_restart.py:51]:
```suggestion
                "WALLTIME": "00:00:05",  # Not enough time to converge an SCF loop
```

File: examples/workchains/example_base_failed_restart.py [hunk 2]
```diff
@@ -110,11 +110,20 @@ def example_base(cp2k_code):
...
-    builder.cp2k.metadata.options.resources = {
-        "num_machines": 1,
-        "num_mpiprocs_per_machine": 1,
+    builder.cp2k.metadata.options = {
+        "resources": {
+            "num_machines": 1,
+            "num_mpiprocs_per_machine": 1,
+        },
+        "max_wallclock_seconds": 1 * 1 * 60,  # 30 min
+        # "withmpi": True,  # A trick to put the kill below before raspa command.
```
Comment (yakutovicha) [inline examples/workchains/example_base_failed_restart.py:119]:
```suggestion
```

File: examples/workchains/example_base_failed_restart.py [hunk 3]
```diff
@@ -110,11 +110,20 @@ def example_base(cp2k_code):
...
+        "mpirun_extra_params": [
+            "timeout",
+            "1",
+        ],  # Kill the calculation after 5 seconds, to test restart.
```
Comment (yakutovicha) [inline examples/workchains/example_base_failed_restart.py:123]:
```suggestion
        ],  # Kill the calculation after 5 seconds to test the restart failure.
```

**Review by yakutovicha — CHANGES_REQUESTED** [review]
Ok, almost good to go. Just a few minor suggestions.

File: examples/workchains/example_base_failed_restart.py [hunk 4]
```diff
@@ -110,12 +109,17 @@ def example_base(cp2k_code):
...
+    builder.cp2k.metadata.options = {
+        "resources": {
+            "num_machines": 1,
+            "num_mpiprocs_per_machine": 1,
+        },
+        "max_wallclock_seconds": 1 * 1 * 60,  # 30 min
+        "mpirun_extra_params": [
+            "timeout",
+            "1",
+        ],  # Kill the calculation after 5 seconds to test the restart failure.
```
Comment (yakutovicha) [inline examples/workchains/example_base_failed_restart.py:121]:
```suggestion
        ],  # Kill the calculation after 1 second to test the restart failure.
```

**Review by yakutovicha — APPROVED** [review]
looks good 👍

**Pull request merged**